### DISCUSSION TITLE
18. SSL 인증서란 무엇인가요?

### DISCUSSION THREAD

**Discussion by eun-byeol** [jewelry-box/cs-interview#20] [General]
- SSL 인증서란 무엇인가요?
- CA(인증 기관)이란 무엇인가요?
- 인증서 체이닝에 대해 설명해 주세요.

**Comment by jongmee** [comment]
- HTTPS에서 사용하는 `디지털 인증서`로 CA의 비밀키로 암호화되어 있어 서버가 `신뢰도 있는 인증 기관에 의해 인증되었음을 사용자에게 보장`할 수 있다. 데이터를 `암호화`해서 전송하기 위해 키를 교환할 때 사용된다.

- SSL 인증서는 `체인 구조`로 되어 있다. 루트 인증서, 중간 인증서, 서버 인증서로 이루어진다. 루트 인증서는 (브라우저에 보통 저장되어 있는) CA의 공개키를 말하고 중간 인증서는 서버 인증서를 신뢰할 수 있게 루트 인증서와 서버 인증서를 이어주는 역할을 한다. 루트 인증서로 서버 인증서를 발급하면 좋겠지만, 루트 CA 서버의 부담이 커져 중간 인증서를 통해 서버 인증서를 발급해준다. 루트 인증서를 보호하고 확장성 있는 구조를 위해 존재하는 체인이다.
